synapse/lib/json.py

```diff
@@ -0,0 +1,191 @@
+import io
+import os
+import mmap
+
+import orjson
+
+import synapse.exc as s_exc
+
+def loads(s):
+    '''
+    Deserialize a JSON string.
+
+    Similar to the standard library json.loads().
+
+    Arguments:
+        s (str | bytes): The JSON data to be deserialized.
+
+    Returns:
+        (object): The deserialized JSON data.
+
+    Raises:
+        synapse.exc.BadJsonText: This exception is raised when there is an error
+            deserializing the provided data.
+    '''
+    try:
+        return orjson.loads(s)
+    except orjson.JSONDecodeError as exc:
+        raise s_exc.BadJsonText(mesg=exc.args[0])
+
+def load(fp):
+    '''
+    Deserialize JSON data from a file.
+
+    Similar to the standard library json.load().
+
+    Arguments:
+        fp (file): The python file pointer to read the data from.
+
+    Returns:
+        (object): The deserialized JSON data.
+
+    Raises:
+        synapse.exc.BadJsonText: This exception is raised when there is an error
+            deserializing the provided data.
+    '''
+    if hasattr(fp, 'mode'):
+        try:
+            with mmap.mmap(fp.fileno(), 0, prot=mmap.PROT_READ) as mm:
+                with memoryview(mm) as mv:
+                    return loads(mv)
+        except ValueError:
+            mesg = 'Cannot read empty file.'
+            raise s_exc.BadJsonText(mesg=mesg)
+    else:
+        return loads(fp.read())
+
+def dumps(obj, sort_keys=False, indent=False, default=None, asbytes=False, append_newline=False):
```

Comment:
Since we're changing these calling conventions, could i propose a few shorter names for these arguments? Perhaps `sort_keys -> sorted` and `append_newline -> newline` and maybe `asbytes` becomes two different functions?

Reply:
- `sort_keys` is from the json stdlib. I left it because it was used in a bunch of places and felt like it would further facilitate a "drop-in ready" solution. We can change it if you like. I'd avoid `sorted` though since that's a python builtin.

- `append_newline` -> `newline` :+1: 

- Not sure I understand what you mean for `asbytes`. Like a `dumpb()` and `dumpsb()` instead of the kwarg?